src/cedalion/sim/synthetic_artifact.py

```diff
@@ -0,0 +1,365 @@
+            trial_type, value, channel).
+        artifacts: Dictionary of artifact functions. Artifact functions must take args
+            (time, onset_time, duration). Keys correspond to the trial_type in the
+            timing DataFrame.
+        mode: 'auto' or 'manual'. If 'auto', artifacts are scaled using the alpha
+            parameter (median of median of sliding windows). If 'manual', artifacts
+            are scaled only by the scale parameter.
+        scale: scaling parameter for artifacts
+
+    Returns:
+        Amplitude data with added artifacts.
+    """
+
+    ts_copy = ts.copy()
+    unit = ts_copy.pint.units if ts_copy.pint.units else 1
+
+    time_start = ts_copy["time"][0].item()
+    time_end = ts_copy["time"][-1].item()
+
+    # set parameters for compute_alpha
+    window_size = ts_copy.time.size // 100 # FIXME window size depends on len(ts_copy)?
+    step_size = window_size // 2
+    channels = ts_copy.channel.values
+
+    # Detect dimension for chromophore or wavelength
+    if "wavelength" in ts_copy.dims:
+        dim_name = "wavelength"
+    elif "chromo" in ts_copy.dims:
+        dim_name = "chromo"
+    else:
+        raise ValueError("No wavelength or chromophore dimension found.")
+    dim_values = ts_copy[dim_name].values
+
+    # generate alpha for each channel/wavelength or channel/chromo
+    if mode == "auto":
+        windows = ts_copy.rolling(time=window_size, center=True).construct(
+            "window", stride=step_size
+        )
+        amplitudes = windows.reduce(np.max, dim="window") - windows.reduce(
+            np.min, dim="window"
+        )
+        alphas = amplitudes.median(dim="time").pint.dequantify()
+    elif mode == "manual":
+        alphas = {
+            (channel, dim_value): scale
+            for dim_value in dim_values
+            for channel in channels
+        }
+    else:
+        raise ValueError("Invalid mode. Must be 'auto' or 'manual'.")
+
+    # make sure events are within bounds of timeseries
+    valid_events = timing[
+        (timing["onset"] >= time_start)
+        & (timing["onset"] + timing["duration"] <= time_end)
+    ]
+
+    for index, row in valid_events.iterrows():
+        onset_time = row["onset"]
+        duration = row["duration"]
+        type = row["trial_type"]
+        sel_channels = row["channel"] if row["channel"] else channels
+        if type in artifacts.keys():
+            artifact = artifacts[type](ts_copy.time, onset_time, duration)
+            for channel in list(set(channels) & set(sel_channels)):
+                for dim_value in dim_values:
+                    alpha = alphas.sel(channel=channel, **{dim_name: dim_value}).item()
+                    ts_copy.loc[dict(channel=channel, **{dim_name: dim_value})] += (
+                        artifact * alpha * scale * unit
+                    )
+        else:
+            print(f"Unknown artifact type {type}")
```

Comment:
Better raise a ValueError instead. This seems to be configuration error that should stop the program.